[14.0][ADD] shopinvader_mass_mailing 📨

ivantodorovich opened the pull request:
Exposes a mailing lists service to allow visitors to subscribe / unsubscribe from them.

ivantodorovich committed:
[ADD] shopinvader_mass_mailing 📨

simahawk commented on shopinvader_mass_mailing/services/mailing_list.py:
```diff
@@ -0,0 +1,256 @@
+
+    _name = "shopinvader.mailing.lists.service"
+    _inherit = "base.shopinvader.service"
+    _usage = "mailing.lists"
```
```suggestion
    _usage = "mailing_lists"
```

simahawk commented on shopinvader_mass_mailing/services/mailing_list.py:
```diff
@@ -0,0 +1,256 @@
+            "name",
+        ]
+        # If the user is authenticated, add the subscription status
+        if self.work.partner_user:
```
```suggestion
        if self._is_logged_in():
```

simahawk commented on shopinvader_mass_mailing/tests/common.py:
```diff
@@ -0,0 +1,33 @@
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```
tracking is disabled on CommonCase setup

simahawk reviewed (commented):
LG overall

ivantodorovich commented:
Done!

Tests are failing due to `openupgradelib` though:
```
  Running command git clone -q https://github.com/OCA/openupgradelib.git /tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d
  Cloning https://github.com/OCA/openupgradelib.git to /tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d
  Resolved https://github.com/OCA/openupgradelib.git to commit 871058afeb6dbd9b34314108b4a8ca6f3e0a9d9f
    ERROR: Command errored out with exit status 1:
     command: /opt/odoo-venv/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/setup.py'"'"'; __file__='"'"'/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-zshxn1wq
         cwd: /tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/setup.py", line 5, in <module>
        import openupgradelib
      File "/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/openupgradelib/__init__.py", line 14, in <module>
        from importlib_metadata import version, PackageNotFoundError
    ModuleNotFoundError: No module named 'importlib_metadata'
```

Seems related to https://github.com/OCA/openupgradelib/pull/243

simahawk commented:
yeah, I got the same issue and reported here https://github.com/shopinvader/odoo-shopinvader/issues/1111 
CC @sbidoul

sbidoul commented:
Please restart test jobs

sbidoul commented:
BTW, this is also a set of REST services that would benefit of being independent of shopinvader (like helpdesk, event, etc).
I think the only abstraction we need is a mechanism to determine the authenticated `res.partner`.
Then these can go into OCA to benefit and attract broader community.

simahawk commented:
I agree, but we could add this to the roadmap for v15.

sbidoul commented:
Ah, yes, v15 :rofl:

sebastienbeau commented:
@PierrickBrun is working on helpdesk case. When he will have a working version he can help on this

simahawk committed:
shopinvader: fix call for public endpoints

When a public endpoint is used, the real use will be 'Public User'
and not the shopinvader tech user.

simahawk committed:
shopinvader_auth_api_key: fix call for public endpoints

When a public endpoint is used, the real use will be 'Public User'
and not the shopinvader tech user.

simahawk committed:
shopinvader_mass_mailing: fix call for public endpoints

When a public endpoint is used, the real use will be 'Public User'
and not the shopinvader tech user.

simahawk commented:
TODO: review sudo changes

ivantodorovich commented:
rebased to solve conflicts

sebastienbeau reviewed (approved):
LGTM (code review)

sebastienbeau commented:
/ocabot merge patch
